happi/tests/test_backends.py

```diff
@@ -203,6 +203,25 @@ def test_json_tempfile_uniqueness():
     assert len(set(tempfiles)) == len(tempfiles)
 
 
+def test_json_tempfile_remove(monkeypatch):
+    # Set consistent temppath
+    jb = JSONBackend("testing.json", initialize=False)
+    temp_path = jb._temp_path()
+    jb._temp_path = lambda: temp_path
+
+    # Ensure file is created, then throw error
+    def shutil_move_patch(*args, **kwargs):
+        assert os.path.isfile(os.path.join(os.getcwd(), temp_path))
```

Comment:
We should be a bit careful here - this assertion might raise, and it'd be caught by `with pytest.raises(BaseException)` below. Assertions are `AssertionError(Exception)` which inherits from `BaseException` (standard library has a nice ASCII tree of this [here](https://docs.python.org/3/library/exceptions.html#exception-hierarchy))

The short of it is: I think you should switch the below clause on line 220 to be `with pytest.raises(RuntimeError)` to make sure we're catching line 215 and not line 214

Reply:
This makes sense, and I'll update the raise. 

I will look more into how assert works in pytest-for some reason I thought the test would just count as a fail and move on. Not sure why I thought of it as isolated